src/main/java/org/opensearch/neuralsearch/ml/MLCommonsClientAccessor.java

```diff
@@ -35,6 +35,33 @@ public class MLCommonsClientAccessor {
     private static final List<String> TARGET_RESPONSE_FILTERS = List.of("sentence_embedding");
     private final MachineLearningNodeClient mlClient;
 
+    /**
+     * Wrapper around {@link #inferenceSentences} that expected a single input text and produces a single floating
+     * point vector as a response.
+     *
+     * @param modelId {@link String}
+     * @param inputText {@link List} of {@link String} on which inference needs to happen
+     * @param listener {@link ActionListener} which will be called when prediction is completed or errored out
+     */
+    public void inferenceSentence(
```

Comment:
you don't need this, when we already have a function which does for a list of sentences, please try to use that.

Reply:
Im going to be writing the code to convert functionality to single sentence/single vector. Why not incorporate it here so it can be used by other features of the code as well?

Reply:
All you need to do is do a get on the vector list, for that if we are creating a new function that is over abstraction. We already have 2 versions of this function in the class, and add third which is very specific seems to be an overkill for now.

Reply:
I prefer to keep this here. I dont see a downside to having it. The name of the method "inferenceSentence" distinguishes it from "inferenceSentences" so there won't be any confusion when to use what. Also, I think single versus collection isnt so specific that it couldn't be useful outside my current use case. Many other Java interfaces/classes have methods for both.

> All you need to do is do a get on the vector list,

True, but you also have to ensure only 1 vector is returned - so each client would also have to add this check/error handling. Using this function, we can centralize that check. Additionally, it is clunky to pass in a List\<List\<Float\>\> listener when you expect that there will only be 1 List\<Float\>

Given that the purpose of this class is to build an easy to use abstraction over MLClient, I think it fits to add this method here.

Reply:
I can add parity method with inferenceSentences where filters are passed in as well.

Reply:
Can we abstract the single sentence function in the queryBuilder class only? I just want to avoid confusion around the usage and 1 more function where we will have use case of not using the TARGET_RESPONSE_FILTERS, for 1 single sentence.

Reply:
I dont think there is any usage confusion, given the method name and signature are descriptive of the functionality.

In terms of maintainability, I dont think it makes sense for other components that will want to use the inferenceSentence method to depend on the queryBuilder class. I think that the ml package should be responsible for providing easy/intuitive interaction with ml-commons for the components in the plugin and should therefore be able to house this functionality. 

That being said, I think it makes sense to either integrate it into this class or build a new class that uses this class to provide higher level abstractions. In a similar case, OpenSearch has a "HighLevel" [rest client](https://github.com/opensearch-project/OpenSearch/blob/main/client/rest-high-level/src/main/java/org/opensearch/client/RestHighLevelClient.java) that is built on top of the lower level RestClient to provide more functionality. We could do this, or we could treat the MLCommonsClientAccessor as the higher level client.